### ISSUE TITLE
[BUG] [CV] missing .detach() before using .numpy()

### ISSUE THREAD

**Issue by JKL98ISR** [deepchecks/deepchecks#1332]
**Describe the bug**
performance check are failing because of missing detach

**To Reproduce**
Run performance check with a model that does something with a gradient on the tensor in some layer 

**Expected behavior**
checks not failing
